### PR TITLE
Changed validation to only allow alpha-numeric names.

### PR DESCRIPTION
**Pull request by jasonallen68**
# Problem space:
- Experiment names with '.', '-', or '_' characters cause the 'experiment' module dependency to throw an error.
 
# Changes:
- [ ] Changed experiment name validation to only allow alpha-numeric characters, where the first character must be an alpha character.  This is more compatible with the 'experiment' module dependency.
- [ ] Updated unit test to validate changes.
- [ ] Changed package.json version to 0.1.0.
 
# Should:
- [ ] Only allow experiment names with alpha-numeric characters.
- [ ] Only allow experiment names to begin with an alpha character.

### CODE REVIEW

**Comment by jasonallen68** [comment]
@redbugz @spacerockzero @lambertsteven I've updated experiment name validation to only allow alpha-numeric characters and the first character must be an alpha character.  This is similar to the validation that the 'experiment' module dependency performs.  Please take a look at this PR if you have some time.

File: test/lib/experiment-validator-test.js [hunk 1]
```diff
@@ -1,16 +1,35 @@
...
+  , validExpNames = [
+    {name: 'ValidExperimentName'},
+    {name: 'Valid1Experiment2Name'}
+  ]
+  , invalidExpNames = [
+    {name: 'Invalid Experiment Name'},
+    {name: '9InvalidExperimentName'},
+    {name: 'Invalid-Experiment-Name'},
+    {name: 'Invalid.Experiment.Name'},
+    {name: 'Invalid_Experiment_Name'}
```
Comment (redbugz) [inline test/lib/experiment-validator-test.js]: would it be useful to have a test for an empty string or a string with just whitespace or invalid characters?

File: test/lib/experiment-validator-test.js [hunk 2]
```diff
@@ -1,16 +1,35 @@
-var assert = require("chai").assert
-  , experimentValidator = require("../../lib/experiment-validator");
+var expect = require("chai").expect
+  , experimentValidator = require("../../lib/experiment-validator")
+  , validExpNames = [
+    {name: 'ValidExperimentName'},
+    {name: 'Valid1Experiment2Name'}
```
Comment (redbugz) [inline test/lib/experiment-validator-test.js]: would it be useful to have a test for the minimal valid string ( a single letter?), are there any length requirements we need to be aware of?

**Comment by redbugz** [comment]
This essentially forces the requirements of fs-webdev/experiment on XPRMTL, which works for us. I guess we'll see if Dan is okay with it.

File: test/lib/experiment-validator-test.js (hunk 2 above)
Comment (jasonallen68) [inline test/lib/experiment-validator-test.js]: I have not found a length requirement.  However, I will add a test for the empty string.

**Comment by jasonallen68** [comment]
@redbugz I added a test for the empty string.

**Comment by redbugz** [comment]
Looks good!
